Setup/UpgradeSchema.php

```diff
@@ -220,6 +220,20 @@ public function upgrade(SchemaSetupInterface $setup, ModuleContextInterface $con
             }
         }
 
+        if (version_compare($context->getVersion(), '2.9.4', '<')) {
```

Comment:
we need another check for version => 3.0.0 && version < 3.0.2

Reply:
It won't work because those are different versions. It will require some adjustments for the CBE version (changing the version).

Reply:
It's possible to do the column change regardless of the version if you prefer (run on every upgrade)

Reply:
i prefer not to , because of the truncation - if you prefer you can check what is the type of the column, and if its float then do the migration. if you will do this check - you don't need to check the version at all.

Reply:
It's also possible. We can check the type on every upgrade & change if needed.
Note that this is not the only setup process that depends on the version (see 2.9.0 on UpgradeData.php & 2.7.5 on UpgradeSchema.php, it's not uncommon to do that).

Reply:
Another option is to switch to a "declarative schema" which is a new way for configuring DB tables on Magento 2 using xml files with no PHP & versioning (introduced on M2.3 & recommended by Magento).

Reply:
as discussed over phone - i prefer not going to declarative now. so lets do the column check then?

Reply:
We did that on branch girit-4.0.0, currently being tested.